server/datastore/mysql/software_installers.go

```diff
@@ -1212,7 +1285,8 @@ WHERE
 			max(id) -- ensure we use only the most recent install attempt for each host
 			FROM host_software_installs
 		WHERE
-			software_title_id = :title_id
+			host_id = hsi.host_id
```

Comment:
That was missing, was that a bug?

Reply:
I think it may have been just potential inefficiency in the prior subquery rather than a bug since the subquery is still grouping by `hsi.host_id` and being then joined on `hosts.id`. This change seemed to improve the `EXPLAIN` when I was testing locally.